imxrt-hal/src/iomuxc/i2c.rs

```diff
@@ -33,193 +35,185 @@ pub trait Pin {
     type Module: module::Module;
 
     /// Perform IOMUXC configurations for this pin
-    fn configure(&mut self);
+    fn configure(self);
```

Comment:
Going from &mut self to self seems like something I should look harder at, if we consume the pad generated by the pad macro, does this mean it is forever an i2c iomuxed pad then?

I guess I'm thinking about low power devices where I want to perhaps dynamically setup and tear down pins/busses/devices. Something to perhaps look harder at in the future, not that concerned while we ramp up here.

Reply:
Good thought about peripheral reconfiguration. Right now, the HAL API doesn't accommodate for any reconfiguration. This pad is forever muxed for I2C purposes. We further enforce this by taking the pads by value in the `i2c::Builder::build()` method. The returned `I2C` object doesn't provide a `release()` method to return the pads. We can figure out how this might change, not only for the pads and peripherals, but also for the clocks that drive the peripherals.